UI/js/controllers.js

```diff
@@ -52,45 +52,64 @@ dashboardApp.factory('Courses', function ($http) {
   };
 });
 
-dashboardApp.controller('coursesController', ['Courses', '$rootScope', '$scope', function (Courses, $rootScope, $scope) {
 
-  $scope.courses = [];
-  $scope.loading = true;
-
-  var url = 'courses/' + $rootScope.user + '.json';
-
-  Courses.getCourses(url).then(function (data) {
-    if (data.failure) {
-      $scope.courses.errors = data;
-      $scope.loading = false;
-    } else {
-      $scope.courses = data;
-      $scope.loading = false;
+dashboardApp.factory('Terms', function ($http) {
```

Comment:
Very few comments in this code - I'd like to see a function header at minimum for long-term maintainability